includes/polling/ospf.inc.php

```diff
@@ -305,30 +307,33 @@
     foreach ($ospf_nbrs_poll as $ospf_nbr_id => $ospf_nbr) {
         // If the entry doesn't already exist in the prebuilt array, insert into the database and put into the array
         if (!isset($ospf_nbrs_db[$ospf_nbr_id][$device['context_name']])) {
-            dbInsert(array('device_id' => $device['device_id'], 'ospf_nbr_id' => $ospf_nbr_id, 'context_name' => $device['context_name']), 'ospf_nbrs');
+            $tmp_insert = array('device_id' => $device['device_id'], 'port_id' => 0, 'ospf_nbr_id' => $ospf_nbr_id, 'context_name' => $device['context_name']);
+            foreach ($ospf_nbr_oids_db as $oid_db) {
+                $tmp_insert[$oid_db] = $ospf_nbr[$oid_db];
+            }
+            dbInsert($tmp_insert, 'ospf_nbrs');
+            unset($tmp_insert);
             echo '+';
             $entry = dbFetchRow('SELECT * FROM `ospf_nbrs` WHERE `device_id` = ? AND `ospf_nbr_id` = ? AND `context_name` = ?', array($device['device_id'], $ospf_nbr_id,$device['context_name']));
             $ospf_nbrs_db[$ospf_nbr_id][$device['context_name']] = $entry;
         }
     }
 
-    if ($debug) {
-        echo "\nPolled: ";
-        print_r($ospf_nbrs_poll);
-        echo 'Database: ';
-        print_r($ospf_nbrs_db);
-        echo "\n";
-    }
+    d_echo("\nPolled: ");
+    d_echo($ospf_nbrs_poll);
+    d_echo('Database: ');
+    d_echo($ospf_nbrs_db);
 
     // Loop array of entries and update
     if (is_array($ospf_nbrs_db)) {
         foreach ($ospf_nbrs_db as $ospf_nbr_id => $ospf_nbr_db) {
+            $ospf_nbr_db = array_shift($ospf_nbr_db);
```

Comment:
This fixes the update issues, but I'm unsure if it breaks users with contexts.  If we are only polling one context at a time, I assume it will be just fine.

Reply:
We poll one context at a time.